etna/models/nn/rnn.py

```diff
@@ -0,0 +1,223 @@
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import Optional
+from typing import TypedDict
+
+import numpy as np
+import pandas as pd
+
+from etna import SETTINGS
+
+if SETTINGS.torch_required:
+    import torch
+    import torch.nn as nn
+
+from etna.models.base import DeepBaseModel
+
+
+class Batch(TypedDict):
+    """Batch specification for RNN."""
+
+    encoder_real: "torch.Tensor"
+    decoder_real: "torch.Tensor"
+    encoder_target: "torch.Tensor"
+    decoder_target: "torch.Tensor"
+    segment: "torch.Tensor"
+
+
+class RNN(DeepBaseModel):
+    """RNN based on LSTM cell."""
+
+    def __init__(
+        self,
+        input_size: int,
+        decoder_length: int,
+        encoder_length: int,
+        num_layers: int = 2,
+        hidden_size: int = 16,
+        lr: float = 1e-3,
+        loss: Optional["torch.nn.Module"] = None,
+        train_batch_size: int = 16,
+        test_batch_size: int = 1,
+        trainer_params: Optional[dict] = None,
+        train_dataloader_params: Optional[dict] = None,
+        test_dataloader_params: Optional[dict] = None,
+        val_dataloader_params: Optional[dict] = None,
+        split_params: Optional[dict] = None,
+        optimizer_params: Optional[dict] = None,
```

Comment:
May be add this parameter to the base class + lr

Reply:
Or place it before the train_bathc_size

Reply:
I'll change order